files/lib/data/conversation/ConversationAction.class.php

```diff
@@ -216,6 +216,11 @@ public function markAsRead() {
 		if (empty($this->parameters['visitTime'])) {
 			$this->parameters['visitTime'] = TIME_NOW;
 		}
+
+		// userID cannot be set during an AJAX request, but only when calling this method within PHP
```

Comment:
> unsetting the userID in the `validateMarkAsRead` method (which handles AJAX requests) and setting in to the currents userID in the `markAsRead` method
This will ensure that the user id can never be set through AJAX, but it will always be present inside `markAsRead()` regardless of the callee. Keep in mind that `validateMarkAsRead()` is only invoked for AJAX-requests, method calls via the PHP API do not cause it to run!

> setting the userID to the currents userID in the `validateMarkAsRead` method
This will not affect calls from PHP, therefore the user id is not set in these cases.